### PR TITLE
ign -> gz CMake, Python, Partial Source, and File Migrations : gz-math

### PR DESCRIPTION
**Pull request by methylDragon**
See: https://github.com/gazebo-tooling/release-tools/issues/765

### CODE REVIEW

**Comment by methylDragon** [comment]
@osrf-jenkins run tests please!

**Comment by methylDragon** [comment]
@osrf-jenkins run tests please!

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 gz_configure_project(
-  REPLACE_IGNITION_INCLUDE_PATH gz/math
+  REPLACE_INCLUDE_PATH gz/math
```
Comment (scpeters) [inline CMakeLists.txt:21]: I think this line can be removed since the default path is `gz/math` as of https://github.com/gazebosim/gz-cmake/pull/277

**Review by scpeters — COMMENTED** [review]
Interval python test failed:

~~~
128: Traceback (most recent call last):
128:   File "/Users/jenkins/workspace/ignition_math-ci-pr_any-homebrew-amd64/ign-math/src/python_pybind11/test/Interval_TEST.py", line 17, in <module>
128:     from ignition.math import Intervald
128: ModuleNotFoundError: No module named 'ignition.math'
128/165 Test #128: Interval_TEST.py ..................................***Failed    0.12 sec
~~~

the python test failures don't show up in homebrew CI unfortunately

**Comment by methylDragon** [comment]
> Interval python test failed:
> 
> ```
> 128: Traceback (most recent call last):
> 128:   File "/Users/jenkins/workspace/ignition_math-ci-pr_any-homebrew-amd64/ign-math/src/python_pybind11/test/Interval_TEST.py", line 17, in <module>
> 128:     from ignition.math import Intervald
> 128: ModuleNotFoundError: No module named 'ignition.math'
> 128/165 Test #128: Interval_TEST.py ..................................***Failed    0.12 sec
> ```
> 
> the python test failures don't show up in homebrew CI unfortunately

New build has it passing: https://build.osrfoundation.org/job/ignition_math-ci-pr_any-homebrew-amd64/1826/consoleFull

There was a bug in how the symlinks were being installed/created. Fixed here and in sim.

File: CMakeLists.txt (hunk 1 above)
Comment (methylDragon) [reply CMakeLists.txt]: Fixed here and everywhere else https://github.com/gazebosim/gz-math/pull/462/commits/008784ae83b8c179317df2a18ec26a7c1bfcabc7

**Review by scpeters — APPROVED** [review]

File: tutorials/installation.md [hunk 2]
```diff
@@ -188,13 +188,13 @@ The optional Eigen component of Gazebo Math requires:
 
   You can view available versions and their dependencies:
   ```
-  conda search libignition-math* --channel conda-forge --info
+  conda search libgz-math* --channel conda-forge --info
   ```
-  See the [Conda release repository](https://github.com/conda-forge/libignition-math4-feedstock) for more information.
+  See the [Conda release repository](https://github.com/conda-forge/libgz-math4-feedstock) for more information.
```
Comment (scpeters) [inline tutorials/installation.md:193]: this conda link doesn't work with the change. I'm not sure if any of the conda portion should be migrated actually

**Review by scpeters — CHANGES_REQUESTED** [review]

File: tutorials/installation.md (hunk 2 above)
Comment (methylDragon) [reply tutorials/installation.md]: Fixed! https://github.com/gazebosim/gz-math/pull/462/commits/bc118cbebbb21007d8e93acd9f0b5490c50baa30

**Comment by methylDragon** [comment]
@osrf-jenkins run tests please!

**Comment by methylDragon** [comment]
@osrf-jenkins run tests please!

**Review by scpeters — APPROVED** [review]

**Pull request merged**